### PR TITLE
Story 16566 create endpoint for deleting customer phone number

### PR DESCRIPTION
**Pull request by SomeSayKosm**
https://invoca.atlassian.net/browse/STORY-16566

I wasn't able to checkout out the [PR](https://github.com/Invoca/developer-docs/pull/341) that will precede this one so I just dropped the page into the RingPool section for now.  It will be moved once the other changes are merged in.

## Checklist

- [ ] Find the [Service owning team](https://docs.google.com/spreadsheets/d/1YF2wuepY5SZTpVEdT9gwhTPhpnDGZeVjywEwvbJS5TQ/edit#gid=0&fvid=1520238175) for these changes, and tag the team as "Reviewers" on this PR
- [ ] Test the documentation changes on readthedocs as a private branch
- [ ] If changing general content, have agreement on whether to apply to latest version or all versions (if all versions, provide links to the related PRs below)

### CODE REVIEW

File: source/api_documentation/network_integration/ringpools/_delete_customer_phone_number.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        "destination_type": "Advertiser",
+        "lifetime_seconds": 300,
+        "max_pool_size": 2,
+        "preferred": false,
+        "is_first_preference": true,
+        "use_custom_data": true
+      },
+      "ignored_at": null,
+      "notified_at": null
+    }
```
Comment (abhig10) [inline source/api_documentation/network_integration/ringpools/_delete_customer_phone_number.rst]: This is delete, so the response body should be `{}`.

File: source/api_documentation/network_integration/ringpools/index.rst [hunk 2]
```diff
@@ -85,6 +85,12 @@ Endpoint:
    :description: Delete a RingPool
    :page: delete_ring_pool
 
+.. api_endpoint::
+   :verb: DELETE
+   :path: /ring_pools/&lt;customer_phone_number&gt;
+   :description: Delete a CustomerPhoneNumber
+   :page: customer_phone_number
```
Comment (abhig10) [inline source/api_documentation/network_integration/ringpools/index.rst]: Path and page is incorrect here. Also, we are calling them Destinations in documentation, unless we want to change it to Customer Phone Number.

```suggestion
   :path: /customer_phone_numbers/&lt;customer_phone_number_id&gt;
   :description: Delete a Destination
   :page: delete_customer_phone_number
```

**Review by abhig10 — CHANGES_REQUESTED** [review]
I'll defer final approval until the time blocking PR's are merged. But until then, we can make the noted fixes.

File: source/api_documentation/network_integration/customer_phone_numbers/index.rst [hunk 3]
```diff
@@ -112,3 +112,9 @@ Endpoint:
    :path: /customer_phone_numbers/&lt;customer_phone_number_id&gt;
    :description: Get a Destination
    :page: get_customer_phone_number
+
+.. api_endpoint::
+   :verb: DELETE
+   :path: /customer_phone_number/&lt;customer_phone_number_id&gt;
+   :description: Delete a Destination
+   :page: delete_customer_phone_number
```
Comment (abhig10) [inline source/api_documentation/network_integration/customer_phone_numbers/index.rst:120]: The path here is incorrect. It should have `customer_phone_numbers`. And also, add a blank line at the bottom.

```suggestion
   :path: /customer_phone_numbers/&lt;customer_phone_number_id&gt;
   :description: Delete a Destination
   :page: delete_customer_phone_number

```

**Review by abhig10 — CHANGES_REQUESTED** [review]
Just one change. Also, since I'm reverting the customer phone number documentation in https://github.com/Invoca/developer-docs/pull/345 and moving them to https://github.com/Invoca/developer-docs/pull/344, you should change the base in PR to `STORY-16669-add-2022-08-01-api-version-for-network-integration-and-add-related-documentation`

**Comment by SomeSayKosm** [comment]
Moving to https://github.com/Invoca/developer-docs/pull/347

**Pull request closed without merging**